server/types/ai.d.ts

```diff
@@ -0,0 +1,26 @@
+import type { GoogleGenerativeAI, InputContent } from '@google/generative-ai';
+
+/**
+ * The interface for the AI.
+ * This will need to modified if we add more AI models.
+ */
+export interface AiInterface {
```

Comment:
The names `AiInterface` and `AiTypes` seem a bit vague. Consider using more descriptive names like AIModelService, AIProviderService, AIModel, AIProvider, or other appropriate names to clearly convey their purposes.

Reply:
I will consider alternative names (including the ones provided). I will add comments here before committing changes.

Reply:
What about:
AiInterface -> GenerativeAIModel
AiTypes -> AIProvider